libraries/AP_HAL_ChibiOS/hwdef/PixPilot-V6PRO/hwdef.dat

```diff
@@ -0,0 +1,285 @@
+
+# MCU class and specific type
+MCU STM32H7xx STM32H743xx
+
+# crystal frequency
+OSCILLATOR_HZ 24000000
+
+# board ID for firmware load
+APJ_BOARD_ID 1160
+
+FLASH_SIZE_KB 2048
+
+
+# with 2M flash we can afford to optimize for speed
+env OPTIMIZE -O2
+
+# bootloader takes first sector
+FLASH_RESERVE_START_KB 128
+
+
+
+# order of I2C buses
+I2C_ORDER  I2C2 I2C1
+
+# order of UARTs (and USB)
+SERIAL_ORDER OTG1 USART2 USART3 UART4 UART8 UART7 OTG2
+
+#DEFAULTGPIO OUTPUT LOW PULLDOWN
+
+# USB.
+PA11 OTG_FS_DM OTG1
+PA12 OTG_FS_DP OTG1
+PA9 VBUS INPUT OPENDRAIN
+PC0 VBUS_nVALID INPUT PULLUP
+
+
+# telem1
+PD5 USART2_TX USART2
+PD6 USART2_RX USART2
+PD3 USART2_CTS USART2
+PD4 USART2_RTS USART2
+
+PD3 EXTERN_GPIO4 OUTPUT GPIO(4) ALT(1)
+PD4 EXTERN_GPIO5 OUTPUT GPIO(5) ALT(1)
+
+# telem2
+PD8 USART3_TX USART3
+PD9 USART3_RX USART3
+PD11 USART3_CTS USART3
+PD12 USART3_RTS USART3
+
+# GPS
+PA0 UART4_TX UART4
+PA1 UART4_RX UART4 NODMA
+
+# GPS2
+PE0 UART8_RX UART8
+PE1 UART8_TX UART8 NODMA
+
+# UART7
+PE7 UART7_RX UART7
+PE8 UART7_TX UART7
+
+# UART for IOMCU
+IOMCU_UART USART6
+PC6 USART6_TX USART6
+PC7 USART6_RX USART6
+
+
+PA2 BATT_VOLTAGE_SENS ADC1 SCALE(1)
+PA3 BATT_CURRENT_SENS ADC1 SCALE(1)
+
+
+
+
+# Now the VDD sense pin. This is used to sense primary board voltage.
+PA4 VDD_5V_SENS ADC1 SCALE(2)
+
+# This defines an output pin which will default to output HIGH. It is
+# a pin that enables peripheral power on this board. It starts in the
+# off state, then is pulled low to enable peripherals in
+# peripheral_power_enable()
+#PA8 nVDD_5V_PERIPH_EN OUTPUT HIGH
+PE3 VDD_3V3_SENSORS_EN OUTPUT LOW
+# SWD debug
+PA13 JTMS-SWDIO SWD
+PA14 JTCK-SWCLK SWD
+
+# PWM output for buzzer
+PA15 TIM2_CH1 TIM2 GPIO(77) ALARM
+
+
+# CAN1
+PD0 CAN1_RX CAN1
+PD1 CAN1_TX CAN1
+
+# CAN2
+PB6 CAN2_TX CAN2
+PB12 CAN2_RX CAN2
+
+# I2C1
+PB8 I2C1_SCL I2C1
+PB9 I2C1_SDA I2C1
+
+# I2C2
+PB10 I2C2_SCL I2C2
+PB11 I2C2_SDA I2C2
+
+
+# SPI1.
+PA5 SPI1_SCK SPI1
+PA6 SPI1_MISO SPI1
+PA7 SPI1_MOSI SPI1
+
+
+# SPI4.
+PE2 SPI4_SCK  SPI4
+PE5 SPI4_MISO SPI4
+PE6 SPI4_MOSI SPI4
+
+# SPI2
+PB13 SPI2_SCK SPI2
+PB14 SPI2_MISO SPI2
+PB15 SPI2_MOSI SPI2
+
+
+# This defines more ADC inputs.
+PC3 AUX_POWER ADC1 SCALE(1)
+PC4 AUX_ADC2 ADC1 SCALE(1)
+
+# And the analog input for airspeed (rarely used these days).
+PC5 PRESSURE_SENS ADC1 SCALE(2)
+
+# More CS pins for more sensors. The labels for all CS pins need to
+# match the SPI device table later in this file.
+PC13 42688_EXT_CS CS
+PD7 BARO_EXT_CS CS
+PC2 40605_EXT_CS CS
+PE4 42688_CS CS
+PC14 BARO_CS CS
+PD10 FRAM_CS CS
+#PD2 SDCARD_CS CS
+
+PC1 42688_EXT_DRDY INPUT
+PC15 40605_EXT_DRDY INPUT
+PD15 42688_DRDY INPUT
+
+# Now we start defining some PWM pins. We also map these pins to GPIO
+# values, so users can set SERVOx_FUNCTION=-1 to determine which
+# PWM outputs on the primary MCU are set up as GPIOs.
+# To match HAL_PX4 we number the GPIOs for the PWM outputs
+# starting at 50.
+PE14 TIM1_CH4 TIM1 PWM(1) GPIO(50)
```

Comment:
I think that output 1-6 can be declared be setup as BIDIR...I tried locally and TIM1 and TIM4 have nonshared DMA so should be possible

Reply:
PE14 TIM1_CH4 TIM1 PWM(1) GPIO(50)
PE13 TIM1_CH3 TIM1 PWM(2) GPIO(51) BIDIR
PE11 TIM1_CH2 TIM1 PWM(3) GPIO(52)
PE9  TIM1_CH1 TIM1 PWM(4) GPIO(53) BIDIR
PD13 TIM4_CH2 TIM4 PWM(5) GPIO(54) BIDIR
PD14 TIM4_CH3 TIM4 PWM(6) GPIO(55) BIDIR

PB0 TIM3_CH3 TIM3 PWM(7) GPIO(56)  NODMA
PB1 TIM3_CH4 TIM3 PWM(8) GPIO(57)  NODMA

Reply:
Have been modified，Thanks.